File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        positives = []
        negatives = []

        for num in nums:
            if num > 0:
                positives.append(num)
            else:
                negatives.append(num)

        for i in range(len(positives)):
            nums[2 * i] = positives[i]
            nums[2 * i + 1] = negatives[i]

        return nums

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int> positives;
        vector<int> negatives;

        for (int num : nums) {
            if (num > 0)
                positives.push_back(num);
            else
                negatives.push_back(num);
        }

        int j = 0;
        for (int i = 0; i < positives.size(); i++) {
            nums[j] = positives[i];
            nums[j + 1] = negatives[i];
            j += 2;
        }

        return nums;
    }
};